spec/battle_boats/engine_spec.rb

```diff
@@ -44,50 +44,116 @@
         coordinate = BattleBoats::Coordinate.new(row: row, column: column)
         invalid_coordinate = BattleBoats::Coordinate.new(row: invalid_row, column: column)
-        error_message = "error"
+        error_message = "ERROR"
         status_report = "STATUS REPORT"
+        ally_status_report = "ALLY STATUS REPORT"
 
         expect(console_ui).to receive(:greet).ordered
-        expect(enemy_board).to receive(:game_over?).and_return(false)
+
+        expect(enemy_board).to receive(:game_over?).and_return(false).ordered
+        expect(ally_board).to receive(:game_over?).and_return(false).ordered
+
+        expect(ally_board).to receive(:status_report).and_return(ally_status_report).ordered
+        expect(console_ui).to receive(:display_status_report).with(ally_status_report).ordered
+        expect(console_ui).to receive(:display_ally_board).with(ally_board).ordered
+
+        expect(enemy_board).to receive(:status_report).and_return(status_report).ordered
+        expect(console_ui).to receive(:display_status_report).with(status_report).ordered
         expect(console_ui).to receive(:display_board).with(enemy_board).ordered
-        expect(console_ui).to receive(:get_coordinate).and_return(invalid_coordinate).ordered
 
+        expect(console_ui).to receive(:get_coordinate).and_return(invalid_coordinate).ordered
         expect(enemy_board).to receive(:strike_position).with(coordinate: invalid_coordinate).and_return(false).ordered
         expect(enemy_board).to receive(:status_report).and_return(error_message).ordered
         expect(console_ui).to receive(:display_status_report).with(error_message).ordered
 
         expect(console_ui).to receive(:get_coordinate).and_return(coordinate).ordered
-
         expect(enemy_board).to receive(:strike_position).with(coordinate: coordinate).and_return(true).ordered
-        expect(enemy_board).to receive(:status_report).and_return(status_report).ordered
-        expect(console_ui).to receive(:display_status_report).with(status_report).ordered
+
+        expect(enemy_board).to receive(:get_random_coordinate).and_return(invalid_coordinate).ordered
+        expect(ally_board).to receive(:strike_position).with(coordinate: invalid_coordinate).and_return(false).ordered
+
+        expect(enemy_board).to receive(:get_random_coordinate).and_return(coordinate).ordered
+        expect(ally_board).to receive(:strike_position).with(coordinate: coordinate).and_return(true).ordered
+
         expect(enemy_board).to receive(:game_over?).and_return(true)
-        expect(console_ui).to receive(:win)
+
+        expect(console_ui).to receive(:display_ally_board).with(ally_board).ordered
         expect(console_ui).to receive(:display_board).with(enemy_board).ordered
 
+        expect(enemy_board).to receive(:game_over?).and_return(true).ordered
+        expect(console_ui).to receive(:win).ordered
+
+        engine.start
+      end
+    end
+
+    context "when the player wins" do
```

Comment:
What is the cost associated with maintaining these tests? 

> Because many collaboration tests use test doubles to specify the contract of a dependency, the cost of refactoring a design is sometimes dramatically increased (which is why Discovery Testing encourages targeted rewrites instead, with refactoring as an exceptional activity). 
-[searls](https://github.com/testdouble/contributing-tests/wiki/London-school-TDD#increased-refactoring-cost)

These tests helped shake loose some collaborator objects, `Board`, and `ConsoleUI`, but I'm afraid now, they're value is waning...

Reply:
Interesting, maybe this is a design signal. Would another intermediate collaborator help shrink these down so they are less necessary?